services/ui-src/src/routes/McparReportPage/McparReportPage.tsx

```diff
@@ -25,10 +32,37 @@ export const McparReportPage = ({ pageJson }: Props) => {
   const previousRoute = findRoute(mcparRoutes, path, "previous", "/mcpar");
   const nextRoute = findRoute(mcparRoutes, path, "next", "/mcpar");
 
-  const onSubmit = () => {
-    // TODO: Wire up submit functionality
+  // get user's state
+  const { user } = useUser();
+  const { state, userRole } = user ?? {};
+  const reportYear = "2022";
+  const reportKey = `${state}${reportYear}`;
+
+  // TODO: get real program name per report
+  const programName = "tempName";
+
+  const onSubmit = async (formData: any) => {
+    if (userRole === UserRoles.STATE_USER) {
+      const report = {
+        key: reportKey,
+        programName: programName,
+        report: formData,
+      };
+      const reportStatus = {
+        key: reportKey,
+        programName: programName,
+        status: "In Progress",
+      };
+      try {
+        await writeReport(report);
+        await writeReportStatus(reportStatus);
+      } catch (error: any) {
+        console.log(WRITE_REPORT_FAILED); // eslint-disable-line
```

Comment:
Did we want a better error handler for this?

Reply:
Good callout! Added an error state. Temporary until we get autosave